dhis-2/dhis-api/src/main/java/org/hisp/dhis/resourcetable/ResourceTableService.java

```diff
@@ -33,50 +33,11 @@
  * @author Lars Helge Overland
  */
 public interface ResourceTableService {
-  /** Generates a resource table containing the hierarchy graph for each OrganisationUnit. */
-  void generateOrganisationUnitStructureTable();
+  /** Generates resource tables. */
+  void generateResourceTables();
```

Comment:
As someone with little analytics table knowledge it's not clear to me why `generateDataApprovalResourceTables` has its own method. Maybe a javadoc would make it clearer? (spread some knowledge for reasons why its separate and whether its's called at different stages of the generate flow)

Reply:
These methods are only called and are only necessary if the "hide unapproved data in analytics" system setting is enabled. They are called from the data approval analytics table manager.

Reply:
Good point on the Javadoc. I will update it in the next PR.

Reply:
lovely 👍